### PR TITLE
KOGITO-2697: Recent saved file is not being show in Recents list

### PR DESCRIPTION
**Pull request by ljmotta**
## Task
https://issues.redhat.com/browse/KOGITO-2697

## Description
On this PR:
- Fix the Recent list
- Code clean up
- Refactor file handling
- Add i18n on Desktop backend

## Demo
ezgif com-video-to-gif

### CODE REVIEW

File: packages/desktop/src/common/File.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@
 export const UNSAVED_FILE_NAME = "unsaved file";
 export const SAMPLE = "sample";
 
-export interface File {
+export interface ElectronFile {
```
Comment (tiagobento) [inline packages/desktop/src/common/File.ts:20]: Can we rename the file to ElectronFile too?

File: packages/desktop/src/backend/FileOperations.ts [hunk 2]
```diff
@@ -57,26 +57,32 @@ export class FileOperations {
...
-      if (data.filePath !== UNSAVED_FILE_NAME) {
-        this.userData.saveFileThumbnail(data.filePath, data.fileType, data.fileContent);
+    ipcMain.on(
+      "savePreview",
+      (event: IpcMainEvent, data: { filePath: string; fileType: string; fileContent: string }) => {
+        console.info(removeFileExtension(data.filePath) + "." + data.fileType);
+        dialog
+          .showSaveDialog(this.window, {
+            defaultPath: removeFileExtension(data.filePath) + "." + data.fileType,
+            title: "Save preview",
```
Comment (tiagobento) [inline packages/desktop/src/backend/FileOperations.ts]: i18n? :D

**Review by tiagobento — CHANGES_REQUESTED** [review]
Great stuff, @ljmotta! Left some minor comments inline.

**Review by tiagobento — APPROVED** [review]
LGTM! Good to merge when green.

**Review by tomasdavidorg — APPROVED** [review]

**Pull request merged**